main_f4.c

```diff
@@ -331,6 +416,19 @@ main(void)
 	/* do board-specific initialisation */
 	board_init();
 
+	/* configure the clock for bootloader activity */
+	rcc_clock_setup_hse_3v3(&clock_setup);
```

Comment:
This is a good point.  By moving this code here the application may be called with the System Clock, uart, a timer and USB configured. This puts the onus on the App to reset and initialize the HW. This could be a problem. So jump_to_app once it determines it will make the jump should revert the SOIC to the state it would have been in like the original code. 

cfin is called on both uart and usb - but they do not use the RCC to reset the peripherals.

The timer is disabled but the RCC is not reset - this could be added.

System clock is not reset back to power on default. - this should be done.